subsys/bluetooth/host/conn.c

```diff
@@ -2245,7 +2245,10 @@ static void deferred_work(struct k_work *work)
 		 * auto connect flag if it was set, instead just cancel
 		 * connection directly
 		 */
-		bt_le_create_conn_cancel();
+		if (bt_le_create_conn_cancel() == -ENOBUFS) {
+			LOG_WRN("No buffers to cancel connection, retrying in 10 ms");
+			k_work_reschedule(dwork, K_MSEC(10));
```

Comment:
Increasing the value of  `CONFIG_BT_CREATE_CONN_TIMEOUT` will have a "terminating condition" as compared to getting into a live-lock using `k_work_reschedule()`. May be we will not face such a live-lock scenario, but lets have a "terminating condition" to catch it. 


```suggestion
			LOG_ERR("No HCI Command buffers to cancel connection");
```

Reply:
There is no "terminating condition" for this `k_work_reschedule` call, and breaks the definition of `BT_CREATE_CONN_TIMEOUT` which will now become minimum timeout for pending LE Create Connection command and no maximum value defined.

Reply:
Isn't that better behavior than what you are suggesting?
If you just give up if the allocation fails the timeout is actually infinite, whereas with the retry at least it will timeout as soon as possible (to the resolution of the retry period).

Reply:
i am not against retry, it is missing a terminate condition.

Reply:
What does the termination condition achieve? If it does stop retrying after N attempts, you are left with a connection permanently in the connecting state, with no future chance to terminate if buffers become available.